src/matrix_solver.py

```diff
@@ -12,8 +12,12 @@ def solve_matrix(A, b):
     """
 
     import numpy as np
-    from numpy.linalg import solve
+    import numpy.linalg as linalg
 
-    temperature_array = solve(A, b)
+    try:
+        i = np.linalg.inv(A)
+        temperature_array = linalg.solve(A, b)
```

Comment:
you should consistently refer to either `np.linalg` or `linalg`.  I suspect both are valid in this code given the way you have imported, but the difference may be confusing to some.